reactor-core/src/main/java/reactor/core/publisher/FluxSink.java

```diff
@@ -113,6 +113,16 @@
 	 */
 	FluxSink<T> onDispose(Disposable d);
 
+	@Override
+	default Flux<T> toFlux() {
+		return Flux.error(new IllegalStateException("A FluxSink does not support back referencing the outer Flux"));
```

Comment:
honestly, I'm not a huge fan of offering an API and then having it error in half the cases. this feels like a step backward towards the same kind of inconsistencies that we had in processors.

Reply:
It's not impossible to support it but its easy to make mistakes if supported. At least its explicit that it doesn't work unlike using Processor directly vs sinks. We can look at the final result once I'm done and see what can go back. Right now i'm aiming for one API entry: Sink/Sinks and will add more from this.

Reply:
fixed in 6f60e9f